src/renderer/qssTooltipPopup/js/qssTooltipPopup.js

```diff
@@ -47,16 +54,43 @@
                 type: "gpii.psp.channelListener",
                 options: {
                     listeners: {
-                        onTooltipUpdated: {
+                        onTooltipUpdated: { // handles the text change on tooltip show
                             func: "{qssTooltipPopup}.updateTooltip",
                             args: ["{arguments}.0"]
+                        },
+                        onTooltipArrowDirection: { // handles the change of the tooltip's arrow direction
+                            func: "gpii.qss.qssTooltipPopup.onTooltipArrowDirection",
+                            args: [
+                                "{arguments}.0",
+                                "{qssTooltipPopup}.model.availableDirections",
+                                "{qssTooltipPopup}.container"
+                            ]
                         }
                     },
                     events: {
-                        onTooltipUpdated: null
+                        onTooltipUpdated: null,
+                        onTooltipArrowDirection: null
```

Comment:
Doesn't seem to be a need for this event - if you are continuing to modelise this in the popup, you can just add gpii.qss.qssTooltipPopup.onTooltipArrowDirection as a model listener

Reply:
Right now we are determining the arrow direction in the main process. That's why we are having this event, which is called when the direction is changed in the main process. And in the renderer this applies the proper styles. So what do you mean by that?